VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -367,4 +378,73 @@ public void orientationChangeTest() {
                 activity.getResources().getConfiguration().orientation);
         assertOrientationSetup(activity);
     }
+
+    /**
+     * Testing creation of various fragments in the {@link CourseUnitNavigationActivity}'s
+     * ViewPager, by supplying its {@link CourseUnitPagerAdapter} with all possible
+     * {@link CourseComponent} models.
+     */
+    @Test
+    public void testFragmentCreations(){
```

Comment:
By the way, this is a unit test, while the requirements seem to call more for an integration test. I have already written the outline of the integration testing at the `initializeTest()` method, which is unfortunately mostly commented out since Robolectric doesn't support view draw callbacks properly, and so the `onPageScrollStateChanged()` callback doesn't get invoked. You could change it to use `onPageSelected()` though I suppose, which should fix this issue in theory. And then just verify that the unit Fragment setup matches that of the mock data that we're using, similar to your approach in writing the tests for parsing courseware models.